src/api/SetTokenAPI.ts

```diff
@@ -152,10 +152,71 @@ export class SetTokenAPI {
     return quantity.mod(naturalUnit).eq(ZERO);
   }
 
+  /**
+   * Calculates the units of components of a Set transferred given an issue or redeem quantity.
+   *
+   * @param  setAddress    Address of the Set
+   * @param  quantity      Quantity of Set issued or redeemed
+   * @return               List of units transferred ordered
+   */
+  public async calculateUnitsTransferred(setAddress: Address, quantity: BigNumber): Promise<BigNumber[]> {
+    this.assertCalculateUnitsTransferred(setAddress, quantity);
+
+    const [naturalUnit, componentUnits] = await Promise.all([
+      this.setToken.naturalUnit(setAddress),
+      this.setToken.getUnits(setAddress),
+    ]);
+
+    return _.map(componentUnits, componentUnit => calculatePartialAmount(componentUnit, quantity, naturalUnit));
```

Comment:
Consider returning an array of objects here that also includes the component address since the order is not clear. It would also be able to be used without another get components call

Reply:
?

Reply:
This should still be done, the single method accesses the units for the component passed in